skyportal/plot.py

```diff
@@ -1032,6 +1032,8 @@ def spectroscopy_plot(
         data.append(df)
     data = pd.concat(data)
 
+    data.sort_values(by=['date_observed', 'wavelength'], inplace=True)
+
```

Comment:
I'm not sure how sorting on wavelength works? I assume each spectrum has a whole array of wavelengths?

Reply:
Yes.  Without including wavelength in the sort, they end up scrambled.

Reply:
I tested having two spectra on the same date, but with different times and it sorted them correctly.  The hover in the spectrum plot only shows date, however, and this should be fixed in a different issue/PR (which I will log).